### ISSUE TITLE
Adding a split method to TF2.0 Record class 

### ISSUE THREAD

**Issue by ra312** [tensorflow/addons#948]
**Describe the feature and the current behavior/state.**
Currently, there is no way to split TFRecord with TF2.0 built-in methods. 
TF.split accepts a tensor. I see no accessible and scalable way to extract Tensors from TFRecord.
**Relevant information**
- Are you willing to contribute it (yes/no):yes
- Are you willing to maintain it going forward? (yes/no): yes
- Is there a relevant academic paper? (if so, where): no paper required, looks straightforward
- Is there already an implementation in another framework? (if so, where):apart from loading datasets from tensorflow-datasets with splitting parameter, I have not seen.
- Was it part of tf.contrib? (if so, where):not sure 

**Which API type would this fall under (layer, metric, optimizer, etc.)**data

**Who will benefit with this feature?** people parsing their raw data to TF.data.Dataset format and productionizing it

**Any other info.**

**Comment by AakashKumarNain** [comment]
Can you provide a toy example?

**Comment by ra312** [comment]
import tensorflow as tf
print(f"tf.__version__={tf.__version__}")

ragged_tensor = tf.ragged.constant([
	[[34,43, 13],[0]],
	[[54,45],[1]],
])
print(f"ragged_tensor_type={ragged_tensor}")
tf_rec = tf.data.TFRecordDataset.from_tensors(ragged_tensor)
print(f"type(tf_rec)={type(tf_rec)}")

**Comment by ra312** [comment]
So I know about tf.split, but the method requires a consistent shape (m,n).

**Comment by ra312** [comment]
Let me see if existing methods allow to split. If not, I will build on top a new method.

**Comment by ra312** [comment]
I found out that using methofd take and skip from tf.data.TFRecordDataset, I can achieve the same functionlaity.
full_dataset = tf.data.TFRecordDataset(input_file)
full_dataset = full_dataset.shuffle()
train_dataset = full_dataset.take(train_size)
test_dataset = full_dataset.skip(train_size)
val_dataset = test_dataset.skip(test_size)
test_dataset = test_dataset.take(test_size)

**Comment by ra312** [comment]
So I would like to close this issue as resolved.
